sdk/storage/azure-storage-file-share/azure/storage/fileshare/_directory_client.py

```diff
@@ -463,6 +463,9 @@ def rename_directory(
         :keyword str file_permission_key:
             Key of the permission to be set for the directory.
             Note: Only one of the file-permission or file-permission-key should be specified.
+        :keyword file_permission_format:
+            Specifies the format in which the permission is returned. Acceptable values are "SDDL" or "binary".
+        :paramtype file_permission_format: str or ~azure.storage.fileshare.models.FilePermissionFormat or None
```

Comment:
- The type of this should not include `FilePermissionFormat`. I believe that is a generated model? Either way, we didn't export it so its not available to customers to use. We should just accept `str`.
- We don't need `None` here. Since it's an optional keyword, its implied.
- Instead of `str`, we should use `Literal` with the two known values. 

Ultimately the type for this everywhere it is used should be `Optional[Literal['SDDL', 'binary']]` but we do not need to include the Optional part in docstrings.